packages/provider/src/transports/wallet-request-handler.ts

```diff
@@ -252,43 +249,72 @@ export class WalletRequestHandler implements ExternalProvider, JsonRpcHandler, P
-
-          let sig = ''
-          // TODO:
-          // if (process.env.TEST_MODE === 'true' && this.prompter === null) {
-          if (this.prompter === null) {
-            // prompter is null, so we'll sign from here
-            sig = await signer.signMessage(ethers.utils.arrayify(message), chainId)
+          let message: any
+
+          // there is a difference in the order of the params:
+          // personal_sign: [data, address]
+          // eth_sign: [address, data]
+          if (request.method === 'personal_sign') {
+            const [data, address] = request.params!
+            message = data
           } else {
-            // prompt user to provide the response
-            sig = await this.prompter.promptSignMessage({ chainId: chainId, message: message })
+            const [address, data] = request.params!
+            message = data
           }
 
-          if (sig && sig.length > 0) {
-            response.result = sig
+          let sig = ''
+
+          let isDeployed: boolean
+          let isUpToDate: boolean
+          if (chainId) {
+            // check if the wallet on required chain is deployed
+            isDeployed = await this.isWalletDeployed(signer, chainId)
+            // if the wallet is deployed, check if the wallet config is up to date
+            if (isDeployed) {
+              isUpToDate = await isWalletUpToDate(signer, chainId)
+            } else {
+              isUpToDate = false
+            }
           } else {
-            // The user has declined the request when value is null
-            throw new Error('declined by user')
+            // if no chainId is provided, we'll assume the wallet is auth chain wallet and is up to date
+            isDeployed = true
+            isUpToDate = true
           }
-          break
-        }
-
-        case 'eth_sign': {
-          // note: message from json-rpc input is in hex format
-          const [signingAddress, message] = request.params!
 
-          let sig = ''
           // TODO:
           // if (process.env.TEST_MODE === 'true' && this.prompter === null) {
           if (this.prompter === null) {
-            // prompter is null, so we'll sign from here
+            // prompter is null, so we'll sign from here if the wallet is deployed and up to date
+            if (!isDeployed) {
+              throw new Error(`WalletRequestHandler: wallet is not deployed for chainId ${chainId}`)
+            }
+            if (!isUpToDate) {
+              throw new Error(`WalletRequestHandler: wallet config is not up to date for chainId ${chainId}`)
+            }
             sig = await signer.signMessage(ethers.utils.arrayify(message), chainId)
           } else {
-            // prompt user to provide the response
-            sig = await this.prompter.promptSignMessage({ chainId: chainId, message: message })
+            // check if wallet is deployed and up to date, if not, prompt user to deploy
```

Comment:
btw, for this block of code, once `promptConfirmWalletDeploy` is executed, will it then ask the user to the sign the message automatically? or will the dapp have to prompt to sign the message again..?

ideally, once it executes the deployment, it will then ask to user the sign the message which initiated this execution

Reply:
yep it does, a few lines below

```
if (promptResultForDeployment) {
    sig = await this.prompter.promptSignMessage({ chainId: chainId, message: message })
}
```